components/brave_education/resources/brave_education.ts

```diff
@@ -0,0 +1,57 @@
+/* Copyright (c) 2024 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+import { EducationPageProxy } from './education_page_proxy'
+import { readCommandMessage } from './message_data_reader'
+
+function pageReady() {
+  return new Promise<void>((resolve) => {
+    if (document.readyState === 'loading') {
+      document.addEventListener('DOMContentLoaded', () => resolve())
+    } else {
+      resolve()
+    }
+  })
+}
+
+async function handleMessage(data: any) {
+  // Read the message provided by the website. If the message is invalid, then
+  // an error will be thrown and will appear on the console for debugging.
+  const message = readCommandMessage(data)
+
+  // Execute the command.
+  const { handler } = EducationPageProxy.getInstance()
+  const { executed } = await handler.executeCommand(message.command);
```

Comment:
this is not the same as the upstream implementation. NTP and What's New are slightly different, but I think what's new is the model to follow here

Reply:
See comments below - I'm happy to file a follow-up!